File: scripts/optimize-images.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const folders = [
  path.join(__dirname, '..', 'assets', 'images', 'team'),
  path.join(__dirname, '..', 'assets', 'images', 'clients')
];

async function convert(dir){
  const files = fs.readdirSync(dir).filter(f=>/(jpg|jpeg|png)$/i.test(f));
  for(const f of files){
    const inPath = path.join(dir,f);
    const outPath = path.join(dir, path.parse(f).name + '.webp');
    try{
      await sharp(inPath).resize({width:800}).webp({quality:80}).toFile(outPath);
      console.log('created', outPath);
    }catch(e){
      console.error('failed', inPath, e.message);
    }
  }
}

(async ()=>{
  for(const d of folders){
    if(fs.existsSync(d)){
      await convert(d);
    }
  }
})();

File: scripts/optimize-images.py
#!/usr/bin/env python3
from PIL import Image
from pathlib import Path

folders = [Path('assets/images/team'), Path('assets/images/clients')]

for folder in folders:
    if not folder.exists():
        continue
    for p in folder.iterdir():
        if p.suffix.lower() in ['.jpg', '.jpeg', '.png']:
            out = p.with_suffix('.webp')
            try:
                im = Image.open(p)
                im.save(out, 'WEBP', quality=80, method=6)
                print('created', out)
            except Exception as e:
                print('error', p, e)

